apis/alloydb/v1alpha1/instance_types.go

```diff
@@ -0,0 +1,195 @@
+	// Example:
+	// instanceTypeRef:
+	//   name: clusterName
+	// For instances of type READ_POOL, set the value using external keyword.
+	// Example:
+	// instanceTypeRef:
+	//   external: READ_POOL
+	// If the instance type is SECONDARY, the delete instance operation does not delete the secondary instance but abandons it instead.
+	// Use deletionPolicy = "FORCE" in the associated secondary cluster and delete the cluster forcefully to delete the secondary cluster as well its associated secondary instance.
+	InstanceTypeRef *refs.AlloyDBClusterTypeRef `json:"instanceTypeRef,omitempty"`
+
+	// Configurations for the machines that host the underlying
+	// database engine.
+	MachineConfig *Instance_MachineConfig `json:"machineConfig,omitempty"`
+
+	// Instance level network configuration.
+	NetworkConfig *Instance_InstanceNetworkConfig `json:"networkConfig,omitempty"`
+
+	// Read pool specific config. If the instance type is READ_POOL,
+	// this configuration must be provided.
+	ReadPoolConfig *Instance_ReadPoolConfig `json:"readPoolConfig,omitempty"`
+}
+
+// AlloyDBInstanceStatus defines the config connector machine state of AlloyDBInstance
+type AlloyDBInstanceStatus struct {
+	/* Conditions represent the latest available observations of the
+	   object's current state. */
+	Conditions []v1alpha1.Condition `json:"conditions,omitempty"`
+
+	// ObservedGeneration is the generation of the resource that was most recently observed by the Config Connector controller. If this is equal to metadata.generation, then that means that the current reported status reflects the most recent desired state of the resource.
+	ObservedGeneration *int64 `json:"observedGeneration,omitempty"`
+
+	// A unique specifier for the AlloyDBInstance resource in GCP.
+	ExternalRef *string `json:"externalRef,omitempty"`
+
+	/* NOTYET
+	// ObservedState is the state of the resource as most recently observed in GCP.
+	ObservedState *AlloyDBInstanceObservedState `json:"observedState,omitempty"`
+	*/
+
+	// Time the Instance was created in UTC.
+	CreateTime *string `json:"createTime,omitempty"`
+
+	// The IP address for the Instance. This is the connection
+	// endpoint for an end-user application.
+	IpAddress *string `json:"ipAddress,omitempty"`
+
+	// The name of the instance resource.
+	Name *string `json:"name,omitempty"`
+
+	// The outbound public IP addresses for the instance. This is available ONLY when
+	// networkConfig.enableOutboundPublicIp is set to true. These IP addresses are used
+	// for outbound connections.
+	OutboundPublicIpAddresses []string `json:"outboundPublicIpAddresses,omitempty"`
+
+	// The public IP addresses for the Instance. This is available
+	// ONLY when networkConfig.enablePublicIp is set to true. This is the
+	// connection endpoint for an end-user application.
+	PublicIpAddress *string `json:"publicIpAddress,omitempty"`
```

Comment:
Done. Cleaned up all the variables in the types so that acronyms are capitalized. Didn't update the corresponding field names in the schema because it'll cause breaking changes.